remotes/mount.go

```diff
@@ -19,45 +19,52 @@ type imageHandler interface {
 	Handle(context.Context, ocischemav1.Descriptor) error
 }
 
-func newImageCopier(ctx context.Context, resolver docker.ResolverBlobMounter, sourceFetcher remotes.Fetcher, targetRepo string) (imageCopier, error) {
+func newImageCopier(ctx context.Context, resolver docker.ResolverBlobMounter, sourceFetcher remotes.Fetcher, targetRepo string, eventNotifier descriptorEventNotifier) (imageCopier, error) {
 	destPusher, err := resolver.Pusher(ctx, targetRepo)
 	if err != nil {
 		return imageCopier{}, err
 	}
 	return imageCopier{
 		sourceFetcher: sourceFetcher,
 		targetPusher:  destPusher,
+		eventNotifier: eventNotifier,
 	}, nil
 }
 
 type imageCopier struct {
 	sourceFetcher remotes.Fetcher
 	targetPusher  remotes.Pusher
+	eventNotifier descriptorEventNotifier
 }
 
 func (h *imageCopier) Handle(ctx context.Context, desc ocischemav1.Descriptor) (err error) {
 	if len(desc.URLs) > 0 {
-		fmt.Fprintf(os.Stderr, "Skipping foreign descriptor %s with media type %s (size: %d)\n", desc.Digest, desc.MediaType, desc.Size)
+		h.eventNotifier(FixupEventTypeSkipDescriptorForeign, &desc, "", nil)
 		return nil
 	}
-	fmt.Fprintf(os.Stderr, "Copying descriptor %s with media type %s (size: %d)\n", desc.Digest, desc.MediaType, desc.Size)
+	h.eventNotifier(FixupEventTypeCopyDescriptorStart, &desc, "", nil)
 	reader, err := h.sourceFetcher.Fetch(ctx, desc)
 	if err != nil {
+		h.eventNotifier(FixupEventTypeCopyDescriptorEnd, &desc, "", err)
 		return err
 	}
 	defer reader.Close()
 	writer, err := h.targetPusher.Push(ctx, desc)
 	if err != nil {
 		if errors.Cause(err) == errdefs.ErrAlreadyExists {
+			h.eventNotifier(FixupEventTypeCopyDescriptorEnd, &desc, "", nil)
 			return nil
```

Comment:
I'm not sure, but it might be nicer to just do `err = nil` here and fall through, WDYT? (in a few places)